Sources/AppcuesKit/Presentation/Traits/Appcues/ExperienceWrapperView.swift

```diff
@@ -11,61 +11,43 @@ import UIKit
 @available(iOS 13.0, *)
 internal class ExperienceWrapperView: UIView {
 
+    var preferredContentSize: CGSize?
+
     let contentWrapperView: UIView = {
         let view = UIView(frame: .zero)
-        view.translatesAutoresizingMaskIntoConstraints = false
         view.clipsToBounds = true
         view.layoutMargins = .zero
         return view
     }()
 
-    var shadowLayer: CAShapeLayer? {
-        willSet {
-            // remove existing
-            shadowLayer?.removeFromSuperlayer()
-        }
-        didSet {
-            if let shadowLayer = shadowLayer {
-                layer.insertSublayer(shadowLayer, at: 0)
-            }
-        }
-    }
+    let shadowWrappingView = UIView()
 
     required init() {
         super.init(frame: .zero)
 
         backgroundColor = .clear
 
-        addSubview(contentWrapperView)
-        positionContentWrapperView()
-
-        NSLayoutConstraint.activate([
-            // this is required so the dialogView has an initial non-zero height, after which it can start sizing to the content.
-            contentWrapperView.layoutMarginsGuide.bottomAnchor.constraint(
-                greaterThanOrEqualTo: contentWrapperView.layoutMarginsGuide.topAnchor,
-                constant: 1)
-        ])
+        addSubview(shadowWrappingView)
+        shadowWrappingView.addSubview(contentWrapperView)
+        positionContentView()
     }
 
     @available(*, unavailable)
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
-    override func layoutSubviews() {
-        super.layoutSubviews()
+    func positionContentView() {
+        contentWrapperView.pin(to: shadowWrappingView)
 
-        if let shadowLayer = shadowLayer {
-            shadowLayer.path = UIBezierPath(
-                roundedRect: contentWrapperView.frame,
-                cornerRadius: contentWrapperView.layer.cornerRadius
-            ).cgPath
-            shadowLayer.shadowPath = shadowLayer.path
-        }
-    }
+        contentWrapperView.translatesAutoresizingMaskIntoConstraints = false
 
-    func positionContentWrapperView() {
         NSLayoutConstraint.activate([
+            // this is required so the dialogView has an initial non-zero height, after which it can start sizing to the content.
+            contentWrapperView.layoutMarginsGuide.bottomAnchor.constraint(
+                greaterThanOrEqualTo: contentWrapperView.layoutMarginsGuide.topAnchor,
+                constant: 1),
```

Comment:
came across this in something I was using earlier as well - is this specifically for the case of being embedded inside of the carousel and needing that initial height for the carousel sizing to work?

Reply:
It's not carousel specific, but if the height is 0, SwiftUI seems to shortcut it layout process and it never gains the proper size.